static/js/editor.ts

```diff
@@ -1,109 +1,88 @@
+   * @param {EditorType} editorType the type of the editor
    */
-  turnIntoEditor: (element: HTMLElement, isReadOnly: boolean, isMainEditor: boolean) => HedyEditor;
-
+  initializeWritableEditor($editor: JQuery, editorType: EditorType, dir?: string): HedyEditor;
+  
   /**
-   * Ininitialize an editor that appears in a modal
-   * @param $editor reference to the div that contains this editor
+   * Initializes a read only editor
+   *
+   * @param {HTMLElement} preview - The element to preview the editor.
+   * @return {HedyEditor} The initialized Hedy editor instance.
    */
-  initializeModalEditor: ($editor: JQuery) => HedyEditor;
+  initializeReadOnlyEditor(preview: HTMLElement, dir?: string): HedyEditor;
 }
 
 export interface HedyEditor {
   /**
- * Represents whether there's an open 'ask' prompt
- */
-  askPromptOpen: boolean;
-  /**
-   * Set the highlither rules for a particular level
-   * @param level      
+   * The contents of the editor
    */
-  setHighliterForLevel: (level: number) => void;
-
+  contents: string;
   /**
-   * @returns the string of the current program in the editor
-   */
-  getValue: () => string;
-
-  /**     
-   * @returns if the editor is set to read-only mode
+   * if the editor is set to read-only mode
    */
-  isReadOnly: () => boolean;
-
-  /**
-   * Sets the editor contents.
-   * @param content the content that wants to be set in the editor
-   */
-  setValue: (content: string) => void;
-
+  isReadOnly: boolean;
   /**
-   * Trim trailing whitespaces
+   * Set the highlither rules for a particular level
+   * @param level
    */
-  trimTrailingSpace: () => void
+  setHighlighterForLevel(level: number): void;
 
   /**
    * Resizes the editor after changing its size programatically
    */
-  resize: () => void;
+  resize(): void;
 
   /**
    * Focuses the text area for the current editor
    */
-  focus: () => void;
+  focus(): void;
 
   /**
    * Clears the errors and annotations in the editor
    */
-  clearErrors: () => void;
+  clearErrors(): void;
 
   /**     
    * Moves to the cursor to the end of the current file
    */
-  moveCursorToEndOfFile: () => void;
+  moveCursorToEndOfFile(): void;
 
   /**
    * Clears the selected text
    */
-  clearSelection: () => void;
+  clearSelection(): void;
 
   /**
   * Removes all breakpoints on the rows.
   **/
-  clearBreakpoints: ()  => void;
-
-  /**
-   * Sets the main editor and also its options
-   */
-  configureMainEditor(): void;
+  clearBreakpoints(): void;
 
   /**
    * Returns the breakpoints as a map-to-css-class
    */
-  getBreakpoints: () => Breakpoints;
-
-  /**
-   * Sets the mode of the editor to read-only or editable depending
-   * on the paramater
-   * @param isReadMode whether the editor will be set to read only mode or not
-   */
-  setEditorMode: (isReadMode: boolean) => void;
+  getBreakpoints(): Breakpoints;
+  // TODO: improve this type definition
+  on(key: any, handler: any): void;
```

Comment:
Guys I don't know how I can put the types that I used in the definition of this function here, can someone help?

Reply:
Thank you so muuuuch!!!! I really tried to look this up, but none of my attempts quite worked. Thank you for the detailed explanation!